ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt

```diff
@@ -72,7 +70,9 @@ class SurveyRepositoryTest : BaseHiltTest() {
   @Test
   fun activateSurvey_alreadyAvailableOffline() =
     runTest(testDispatcher) {
-      surveyStore.insertOrUpdateSurvey(SURVEY).await()
+      fakeRemoteDataStore.surveys = listOf(SURVEY)
+      surveyRepository.syncSurveyWithRemote(SURVEY.id).await()
+      advanceUntilIdle()
```

Comment:
I think one `advanceUntilIdle()` is sufficient.

Would moving the setup to `@Before` break other tests?

Reply:
Yes, we don't want to sync everything in @Before. Some tests want to invoke that by activating the survey manually.